src/main/java/ch/njol/skript/expressions/ExprUnleashReason.java

```diff
@@ -0,0 +1,69 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Events;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.EventValueExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.event.Event;
+import org.bukkit.event.entity.EntityUnleashEvent;
+import org.bukkit.event.entity.EntityUnleashEvent.UnleashReason;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Unleash Reason")
+@Description("The unleash reason in an unleash event.")
+@Examples({
+	"if the unleash reason is distance:",
+		"\tbroadcast \"The leash was snapped in half.\""
+})
+@Events("Unleash")
+@Since("INSERT VERSION")
+public class ExprUnleashReason extends EventValueExpression<UnleashReason> {
+
+	public ExprUnleashReason() {
+		super(UnleashReason.class);
+	}
+
+	static {
+		Skript.registerExpression(ExprUnleashReason.class, EntityUnleashEvent.UnleashReason.class, ExpressionType.SIMPLE, "[the] unleash reason");
+	}
+
+	@Override
+	public boolean init(Expression<?>[] expressions, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (!getParser().isCurrentEvent(EntityUnleashEvent.class)) {
+			Skript.error("The 'unleash reason' expression can only be used in an 'unleash' event");
+			return false;
+		}
+		return true;
+	}
+
+	@Override
+	protected UnleashReason[] get(Event event) {
+		if (!(event instanceof EntityUnleashEvent))
+			return new UnleashReason[0];
+		return CollectionUtils.array(((EntityUnleashEvent) event).getReason());
```

Comment:
```suggestion
		if (!(event instanceof EntityUnleashEvent unleashEvent))
			return new UnleashReason[0];
		return new UnleashReason[]{unleashEvent.getReason()};
```
To expand on Efnilite's suggestion